src/metrics.ts

```diff
@@ -0,0 +1,143 @@
+import { Weapon } from "./weapon";
+
+export enum MetricPath {
+  WINDUP_HORIZONTAL = "attacks.horizontal.light.windup",
+  WINDUP_OVERHEAD = "attacks.overhead.light.windup",
+  WINDUP_STAB = "attacks.stab.light.windup",
+  WINDUP_SPECIAL = "specialAttack.windup",
+
+  RANGE_HORIZONTAL = "attacks.horizontal.range",
+  RANGE_ALT_HORIZONTAL = "attacks.horizontal.altRange",
+  RANGE_OVERHEAD = "attacks.overhead.range",
+  RANGE_ALT_OVERHEAD = "attacks.overhead.altRange",
+  RANGE_STAB = "attacks.stab.range",
+  RANGE_ALT_STAB = "attacks.stab.altRange",
+  // RANGE_SPECIAL = "specialAttack.range", TODO
+
+  DAMAGE_HORIZONTAL_LIGHT = "attacks.horizontal.light.damage",
+  DAMAGE_HORIZONTAL_HEAVY = "attacks.horizontal.heavy.damage",
+  DAMAGE_OVERHEAD_LIGHT = "attacks.overhead.light.damage",
+  DAMAGE_OVERHEAD_HEAVY = "attacks.overhead.heavy.damage",
+  DAMAGE_STAB_LIGHT = "attacks.stab.light.damage",
+  DAMAGE_STAB_HEAVY = "attacks.stab.heavy.damage",
+  DAMAGE_SPECIAL = "specialAttack.damage"
+}
+
+export enum MetricLabel {
+  WINDUP_HORIZONTAL = "Windup - Horizontal",
+  WINDUP_OVERHEAD = "Windup - Overhead",
+  WINDUP_STAB = "Windup - Stab",
+  WINDUP_SPECIAL = "Windup - Special",
+  WINDUP_AVERAGE = "Windup - Average*",
+
+  RANGE_HORIZONTAL = "Range - Horizontal",
+  RANGE_ALT_HORIZONTAL = "Range - Alt. Horizontal",
+  RANGE_OVERHEAD = "Range - Overhead",
+  RANGE_ALT_OVERHEAD = "Range - Alt. Overhead",
+  RANGE_STAB = "Range - Stab",
+  RANGE_ALT_STAB = "Range - Alt. Stab",
+  // RANGE_SPECIAL = "Range - Special", TODO
+  RANGE_AVERAGE = "Range - Average*",
+
+  DAMAGE_HORIZONTAL_LIGHT = "Damage - Horizontal (Light)",
+  DAMAGE_HORIZONTAL_HEAVY = "Damage - Horizontal (Heavy)",
+  DAMAGE_OVERHEAD_LIGHT = "Damage - Overhead (Light)",
+  DAMAGE_OVERHEAD_HEAVY = "Damage - Overhead (Heavy)",
+  DAMAGE_STAB_LIGHT = "Damage - Stab (Light)",
+  DAMAGE_STAB_HEAVY = "Damage - Stab (Heavy)",
+  DAMAGE_SPECIAL = "Damage - Special",
+  DAMAGE_AVERAGE = "Damage - Average*",
```

Comment:
These are the human readable labels for the stats we will produce.  Note its mostly 1:1 with MetricPath, but this also has averages (because we can aggregate stats).